client/src/main/java/org/apache/cassandra/sidecar/client/exception/RetriesExhaustedException.java

```diff
@@ -28,24 +29,52 @@ public class RetriesExhaustedException extends RuntimeException
     /**
      * Constructs an exception with the number of {@code attempts} performed for the request.
      *
-     * @param attempts the number of attempts performed for the request
-     * @param request  the HTTP request
+     * @param attempts      the number of attempts performed for the request
+     * @param request       the HTTP request
+     * @param lastResponse  the last failed HTTP response
      */
-    public RetriesExhaustedException(int attempts, Request request)
+    public static RetriesExhaustedException of(int attempts,
+                                               Request request,
+                                               HttpResponse lastResponse)
     {
-        this(attempts, request, null);
+        return of(attempts, request, lastResponse, null);
```

Comment:
call the ctor directly?

Reply:
I feel those static creation methods make the code look nicer by eliminating the need to explicitly pass null.

Reply:
I mean, it's one additional call in the stack, but it might get optimized by the jvm.